### PR TITLE
Expose LocalEndPoint property on SocketListener

### PR DESCRIPTION
**Pull request by stbraley**
When using a system assigned port for the SocketListener. The SocketListener class needs to expose the address of the socket so the caller can obtain the assigned port. I've added a property to expose the endpoint of the socket. If the socket is null, then the property will return a empty endpoint.

### CODE REVIEW

**Comment by davidfowl** [comment]
It makes sense to expose the entire socket.

File: src/System.IO.Pipelines.Networking.Sockets/SocketListener.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ namespace System.IO.Pipelines.Networking.Sockets
     public class SocketListener : IDisposable
     {
         private readonly bool _ownsFactory;
-        private Socket _socket;
+        public Socket _socket;
```
Comment (davidfowl) [inline src/System.IO.Pipelines.Networking.Sockets/SocketListener.cs]: Not like that, expose a readonly property...

**Comment by davidfowl** [comment]
I read this incorrectly. I though you wanted to expose the client socket...

**Comment by stbraley** [comment]
No, need to get the listening port number.

Modeling after the TcpListener in the core framework.

`var listener = new SocketListener();
 var endPoint = new IPEndPoint(IPAddress.Loopback, 0);
_socketListener.Start(_endPoint);

var port = ((IPEndPoint)listener.LocalEndpoint).Port;`

File: src/System.IO.Pipelines.Networking.Sockets/SocketListener.cs [hunk 2]
```diff
@@ -86,6 +86,8 @@ public void Stop()
             }
         }
 
+        public Socket ListeningSocket { get { return _socket;} }
```
Comment (davidfowl) [inline src/System.IO.Pipelines.Networking.Sockets/SocketListener.cs]: C#6 ify

```C#
public Socket ListeningSocket => _socket;
```

**Comment by davidfowl** [comment]
LGTM

**Pull request merged**